tests/functional/conftest.py

```diff
@@ -5,6 +5,8 @@
 import pytest
 from webtest import TestApp
 
+import tests.common.factories
```

Comment:
You can just make `tests/functional/` a package if you want.

> aren't relative imports bad anyway?

No? The form of "relative import" that can cause all sorts of problems is the _implicit_ relative import (i.e. `import foo` where `foo` is a subpackage of the current package), not _explicit_ relative imports, which are absolutely fine.

Putting the import inside the fixture function is absolutely fine, though. Yes, it helps avoid a name conflict, but it also means we don't actually import the factories module unless somebody uses the factories fixture. In application code this pattern should be used sparingly (IMO) but in test code it's absolutely fine, and can actually lead to a cleaner `conftest.py` (which typically imports all kinds of different modules in order to support various fixtures).